### PR TITLE
add in blanket error page

### LINKED ISSUE #83: app should handle gracefully any http errors, including backend being down entirely

### PR DESCRIPTION
**Pull request by rennerpa93**
closes #83 

Created Blanket Error page to serve to user when APIs fail.

### CODE REVIEW

File: angular/e2e/src/testers/custom-equalitytester.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function CustomEqualityTester(testStr: string, testStrArray: string[]): boolean {
```
Comment (DevJamesC) [inline angular/e2e/src/testers/custom-equalitytester.ts:1]: Expect array to contain encountered url.

File: angular/src/app/services/globalerrorhandler/globalerrorhandler.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Injectable, ErrorHandler, Injector } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GlobalerrorhandlerService implements ErrorHandler {
```
Comment (fredbelotte) [inline angular/src/app/services/globalerrorhandler/globalerrorhandler.service.ts:8]: follow casing pattern, PascalCasing for the service and kebab-casing for the file.

File: angular/src/app/app.module.ts [hunk 3]
```diff
@@ -6,16 +6,18 @@ import { Monitoring } from './data/monitoring.model';
...
   bootstrap: [AppComponent],
-  declarations: [AppComponent],
+  declarations: [AppComponent, ErrorComponent],
   imports: [AppRoutingModule, BrowserModule, LayoutModule],
   providers: [
     Monitoring,
     {
       provide: ErrorHandler,
-      useClass: environment.production ? MonitoringService : ErrorHandler,
+      useClass: environment.production ? MonitoringService : GlobalerrorhandlerService,
```
Comment (fredbelotte) [inline angular/src/app/app.module.ts:20]: let's research the ability to add multiple error handlers.

File: angular/e2e/src/testers/custom-equalitytester.ts (hunk 1 above)
Comment (fredbelotte) [inline angular/e2e/src/testers/custom-equalitytester.ts:1]: remove this file and change the jasmine/expect to array.contains(url) with expect being true.

File: angular/e2e/src/lodging.e2e-spec.ts [hunk 4]
```diff
@@ -1,4 +1,9 @@
 import { browser, logging, WebDriver, by, element } from 'protractor';
+import { CustomEqualityTester } from './testers/custom-equalitytester';
+
+beforeEach(() => {
+  jasmine.addCustomEqualityTester(CustomEqualityTester);
```
Comment (fredbelotte) [inline angular/e2e/src/lodging.e2e-spec.ts:5]: no longer need with the change to array.contains(url).

**Review by fredbelotte — COMMENTED** [review]
lgtm after environment change to prod

**Review by fredbelotte — APPROVED** [review]

**Pull request merged**